File: day1.rb
lines = File.readlines('day1_input.txt', chomp: true)

def part_1 lines
  left = []
  right = []
  lines.each do |line|
    left << line.split(" ")[0].to_i
    right << line.split(" ")[1].to_i
  end

  left = left.sort
  right = right.sort

  running_sum = 0
  puts left
  # numbers.zip(letters, colors).each do |num, let, col|

  (0...left).to_a.zip(left, right).each do |num, left_num, right_num|
    puts num
    running_sum += (left_num - right_num).abs
    # puts (left_num)
  end 
  # puts running_sum
end

part_1 lines

File: day1.py
# with open("filename.txt", "r") as file:
#     content = file.read()
#     print(content)

import re

def read_in_data(input_data):
  with open(input_data, "r") as file:
    return file.readlines()

input_data = read_in_data("day1_input.txt")

def part_one():
  left = []
  right = []
  for line in input_data:
    left.append(int(line.strip().split()[0]))
    right.append(int(line.strip().split()[1]))

  left = sorted(left)
  right = sorted(right)

  difference = 0
  for i in range(len(left)):
    difference += abs(left[i] - right[i])

  return difference, set(left), right

def part_two():
  left_set = part_one()[1]
  right = part_one()[2]
  count = 0
  running_sum = 0
  counter = dict()
  for val in right:
    if not val in counter:
      counter[val] = 1
    else:
      counter[val] += 1

  for val in left_set:
    if val in counter:
      running_sum += val * counter[val] 

  return running_sum
print(part_two())

